src/components/course-settings/delete-period.cjsx

```diff
@@ -0,0 +1,74 @@
+React = require 'react'
+BS = require 'react-bootstrap'
+_ = require 'underscore'
+{PeriodActions, PeriodStore} = require '../../flux/period'
+{RosterActions, RosterStore} = require '../../flux/roster'
+{TutorInput} = require '../tutor-input'
+
+EMPTY_WARNING = 'Only periods without students enrolled can be deleted.'
+
+module.exports = React.createClass
+  displayName: 'DeletePeriodLink'
+  propTypes:
+    courseId: React.PropTypes.string.isRequired
+    periods: React.PropTypes.array.isRequired
+    activeTab: React.PropTypes.object.isRequired
+    reloadCourse: React.PropTypes.func.isRequired
+
+  getInitialState: ->
+    warning: ''
+
+  performUpdate: ->
+    if @isPeriodEmpty()
+      @refs.overlay.hide()
+      id = @props.activeTab.id
+      PeriodActions.delete(id)
+      @props.reloadCourse()
```

Comment:
Why does the whole course need to be reloaded here? Could the function be called in the Period flux store inside `.deleted: (periodId) ->` ?